platform/posix/transport/src/openssl_posix.c

```diff
@@ -590,37 +590,31 @@ OpensslStatus_t Openssl_Connect( NetworkContext_t * pNetworkContext,
 
 OpensslStatus_t Openssl_Disconnect( NetworkContext_t * pNetworkContext )
 {
-    OpensslStatus_t returnStatus = OPENSSL_SUCCESS;
-    SocketStatus_t socketStatus = SOCKETS_SUCCESS;
+    SocketStatus_t socketStatus = SOCKETS_INVALID_PARAMETER;
```

Comment:
Using `SOCKETS_INVALID_PARAMETER` as the initialized value carries the knowledge that the `convertToOpensslStatus` translates `SOCKETS_INVALID_PARAMETER` to `OPENSSL_INVALID_PARAMETER`. 
If the translation would change (though quite unlikely), then the return value of this function for an invalid `pNetworkContext` would change..

Reply:
I have thought about it too, but I think it is  fair assumption to make that SOCKETS_INVALID_PARAMETER is converted to OPENSSL_INVALID_PARAMETER always. I have added comments to clarify that